x-pack/test/reporting/functional/lib/compare_pdfs.js

```diff
@@ -73,52 +73,45 @@ export async function checkIfPdfsMatch(actualPdfPath, baselinePdfPath, screensho
   // don't want to start causing failures for other devs working on OS's which are lacking snapshots.  We have
   // mac and linux covered which is better than nothing for now.
   try {
-    await writeFileAsync(baselineCopyPath, fs.readFileSync(baselinePdfPath));
+    log.debug(`writeFileSync: ${baselineCopyPath}`);
+    fs.writeFileSync(baselineCopyPath, fs.readFileSync(baselinePdfPath));
   } catch (error) {
     log.error(`No baseline pdf found at ${baselinePdfPath}`);
     return 0;
   }
-  await writeFileAsync(actualCopyPath, fs.readFileSync(actualPdfPath));
+  log.debug(`writeFileSync: ${actualCopyPath}`);
+  fs.writeFileSync(actualCopyPath, fs.readFileSync(actualPdfPath));
 
   const convertOptions = {
     '-density': '300',
   };
   const actualPdfImage = new PDFImage(actualCopyPath, { convertOptions });
   const expectedPdfImage = new PDFImage(baselineCopyPath, { convertOptions });
 
-  let pageNum = 0;
+  log.debug(`Calculating numberOfPages`);
+
+  const actualDoc = await PDFJS.getDocument(actualCopyPath);
+  const expectedDoc = await PDFJS.getDocument(baselineCopyPath);
+  const actualPages = actualDoc.numPages;
+  const expectedPages = expectedDoc.numPages;
+
+  if (actualPages !== expectedPages) {
+    throw new Error(
+      `Expected ${expectedPages} pages but got ${actualPages} in PDFs expected: "${baselineCopyPath}" actual: "${actualCopyPath}".`
+    );
+  }
+
   let diffTotal = 0;
-  // Ran across an instance where the page conversion failed with `Failed to convert page to image` for no known
-  // reason. Seeing if a loop will resolve these flaky errors.
-  let failCount = 0;
-  while (true) {
-    let expectedPagePng;
-    let actualPagePng;
-    try {
-      log.debug(`Converting expected pdf page ${pageNum} to png`);
-      expectedPagePng = await expectedPdfImage.convertPage(pageNum);
-      log.debug(`Converting actual pdf page ${pageNum} to png`);
-      actualPagePng = await actualPdfImage.convertPage(pageNum);
-    } catch (e) {
-      log.error(`Error caught while converting pdf page ${pageNum} to png: ${e.message}`);
-      if (JSON.stringify(e).indexOf('Requested FirstPage is greater than the number of pages in the file') >= 0) {
-        break;
-      } else {
-        if (failCount < 3) {
-          log.error(`${failCount}: Will try conversion again...`);
-          failCount++;
-          continue;
-        } else {
-          log.error(`Failed ${failCount} times, throwing error`);
-          throw e;
-        }
-      }
-    }
 
+  for (let pageNum = 0; pageNum <= expectedPages; ++pageNum) {
```

Comment:
Just FYI, I think we need to bring back the retry loop.  Just hit the "Failed to convert page to image" error in a PR here: https://github.com/elastic/kibana/pull/24109#issuecomment-430683184